### PR TITLE
LPS-195577

### PR DESCRIPTION
**Pull request by danielCamara02**
Passed on CI in this run: https://github.com/danielCamara02/liferay-portal/pull/58

Ticket: https://liferay.atlassian.net/browse/LPS-195577

### CODE REVIEW

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/SearchFacets.testcase [hunk 1]
```diff
@@ -1090,6 +1090,89 @@ definition {
...
+
+			Refresh();
+		}
+
+		task ("Assert the sort order is changed from 'Term Frequency Descending' to 'Term Frequency Ascending'") {
+			for (var categories : list "Engineering,Product,Sales,Community") {
+				AssertElementPresent(
+					categoryName = ${categories},
+					locator1 = "Search#CATEGORY_FACET_VOCABULARY_LAYOUT");
+			}
```
Comment (xbrianlee) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/SearchFacets.testcase:1160]: Order matters here. This assertion does not care about order. Can we use `SearchFacetPortlet.viewFacetItemsSpecificOrder`?

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/SearchFacets.testcase [hunk 2]
```diff
@@ -1090,6 +1090,89 @@ definition {
...
+			SearchFacetPortlet.updateFacetConfiguration(
+				portletName = "Category Facet",
+				sortOption = "Term Frequency Ascending");
+
+			Refresh();
+
+			for (var categories : list "Community,Sales,Product,Engineering") {
+				AssertElementPresent(
+					categoryName = ${categories},
+					locator1 = "Search#CATEGORY_FACET_VOCABULARY_LAYOUT");
```
Comment (xbrianlee) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/SearchFacets.testcase:1171]: Same comment as above

File: portal-web/test/functional/com/liferay/portalweb/macros/SearchFacetPortlet.macro [hunk 3]
```diff
@@ -350,6 +350,17 @@ definition {
 					value1 = ${facetItem});
 			}
 		}
+
+		if (${portletName} == "Category Facet") {
```
Comment (xbrianlee) [inline portal-web/test/functional/com/liferay/portalweb/macros/SearchFacetPortlet.macro:354]: Follow the following paradigm:
```
if (X) {
	//Do something
}
else if (Y) {
	//Do something else
}
else {
	//Do yet another thing
}
```

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/SearchFacets.testcase [hunk 4]
```diff
@@ -1090,6 +1090,85 @@ definition {
 		}
 	}
 
+	@description = "This is a use case for LPS-171290 - When configuring the Category Facet with the vocabulary layout the different ordering options don't take effect."
+	@priority = 3
+	test CanChangeSortOrderOfCategoriesUsingVocabularyLayout {
+		task ("Create Vocabulary and Categories") {
+			JSONCategory.addVocabulary(
+				groupName = "Guest",
+				title = "V");
```
Comment (xbrianlee) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/SearchFacets.testcase:1099]: Rename "Vocabulary Name" to be more descriptive and for parity with other tests

**Comment by xbrianlee** [comment]
@danielCamara02 Please make the changes above and then forward your changes if they pass testing. Thanks!

**Comment by danielCamara02** [comment]
ci:forward

**Pull request closed without merging**